sentinelhub/download/client.py

```diff
@@ -139,31 +139,37 @@ def _is_download_required(self, request, response_path):
                (self.redownload or response_path is None or not os.path.exists(response_path))
 
 
-def get_json(url, post_values=None, headers=None, download_client_class=DownloadClient):
+def get_json(url, post_values=None, headers=None, request_type=None, download_client_class=DownloadClient, **kwargs):
```

Comment:
Yes, I totally agree that it would be better to mimic `requests` parameters. But I made a mistake of naming parameters differently a few years ago and since then we used this function in quite a lot of places in our codebase. Not to mention that other users probably also used it.

Because I fear this might require a lot of changes I wouldn't go changing/deprecating this now but maybe in some future iteration.

Reply:
Not to mention that Python not enforcing private/protected functions brings additional backward + "what if users use this method" pain. 

At some point refactoring just has to happen. I am ok with pushing this one to (next/future) iteration, but I am also afraid that we already have a lot of such cases, which then just end up not being fixed ever.